libsolidity/analysis/DeclarationContainer.cpp

```diff
@@ -115,3 +115,57 @@ std::vector<Declaration const*> DeclarationContainer::resolveName(ASTString cons
 		return m_enclosingContainer->resolveName(_name, true);
 	return vector<Declaration const*>({});
 }
+
+vector<ASTString> DeclarationContainer::similarNames(ASTString const& _name) const
+{
+	vector<ASTString> similar;
+
+	for (const auto & declaration: m_declarations)
+	{
+		const string& declarationName = declaration.first;
+		if (DeclarationContainer::areSimilarNames(_name, declarationName))
+			similar.push_back(declarationName);
+	}
+
+	if (m_enclosingContainer)
+	{
+		vector<ASTString> enclosingSimilar = m_enclosingContainer->similarNames(_name);
+		similar.insert(similar.end(), enclosingSimilar.begin(), enclosingSimilar.end());
+	}
+
+	return similar;
+}
+
+bool DeclarationContainer::areSimilarNames(ASTString const& name1, ASTString const& name2)
```

Comment:
I would move this to `libdevcore` and name it something like `stringWithinDistance(name1, name2, distance)`. It can be later reused for showing potential names for visibility specifiers, etc.